### PR TITLE
fix(control): prevent negative cancellation fees in order and event cancellation forms

### LINKED ISSUE #5373: Fix: Negative values accepted in "Keep a fixed cancellation fee per ticket" on event cancellation page
### Description
On the Event Cancellation page (`/control/event/<organizer>/<event>/cancel/`), the input field **"Keep a fixed cancellation fee per ticket"** lacks a minimum value constraint (`min_value=Decimal('0.00')`). As a result, organizers can enter and submit negative values without encountering any form validation errors.

---

### Exact Location
* **Web UI Path**: 
  - **URL**: `http://localhost:8000/control/event/<organizer>/<event>/cancel/`
  - **Navigation**: Event Dashboard $\rightarrow$ **Home** $\rightarrow$ **Event settings** $\rightarrow$ Scroll to **Danger zone** $\rightarrow$ Click **"Cancel event"** $\rightarrow$ Section: **Refund options** $\rightarrow$ Field: **"Keep a fixed cancellation fee per ticket"**
* **Code Location**: 
  - `app/eventyay/control/forms/orders.py`
  - Class: `EventCancelForm`
  - Field: `keep_fee_per_ticket` (Line 674)

---

### Steps to Reproduce
1. Open any event in the Control Panel.
2. Navigate to `/control/event/<organizer>/<event>/cancel/`.
3. Under the **Refund options** section, find the field **"Keep a fixed cancellation fee per ticket"**.
4. Enter a negative amount (e.g. `-5.00`).
5. Submit the cancellation form.
6. **Observed Behavior:** The form accepts the negative value without raising a validation error.

---

### Expected Behavior
- The form should enforce `min_value=Decimal('0.00')` and reject negative values, showing an inline validation error: `"Ensure this value is greater than or equal to 0.00."`.

---

### Proposed Fix
- Add `min_value=Decimal('0.00')` to `keep_fee_per_ticket` in `EventCancelForm` (`app/eventyay/control/forms/orders.py`).
- Add automated unit tests to verify that negative inputs are rejected.

### Demo Video

https://github.com/user-attachments/assets/690eefa8-6552-43d3-af65-4518cc25c25a

### PR DESCRIPTION
**Pull request by anuragverma025**
### Description
Fixes #5373

Enforces non-negative bounds on cancellation fee input fields across order and event cancellation forms to prevent organizers from entering negative amounts.

---

### Changes Made
1. **`CancelForm` (`app/eventyay/control/forms/orders.py`)**:
   - Added `min_value=Decimal('0.00')` to `cancellation_fee` DecimalField and `__init__`.
   - Added validation check in `clean_cancellation_fee()` to reject negative values with a clear validation error.

2. **`EventCancelForm` (`app/eventyay/control/forms/orders.py`)**:
   - Added `min_value=Decimal('0.00')` to `keep_fee_per_ticket` and `keep_fee_fixed`.
   - Added `min_value=Decimal('0.00')` and `max_value=Decimal('100.00')` to `keep_fee_percentage`.

3. **Automated Unit Tests (`app/tests/tickets/control/test_orders.py`)**:
   - Added `test_order_cancel_negative_fee`: Verifies that submitting a negative cancellation fee fails form validation and prevents order cancellation.
   - Added `test_event_cancel_form_negative_fees`: Verifies that `EventCancelForm` rejects negative amounts for `keep_fee_fixed`, `keep_fee_per_ticket`, and `keep_fee_percentage`.

---

### Verification & Testing
- Ran automated test suite via Docker:
  ```bash
  pytest tests/tickets/control/test_orders.py::test_order_cancel_negative_fee tests/tickets/control/test_orders.py::test_event_cancel_form_negative_fees

### Screenshots
| After (Validation enforced) |
<img width="1578" height="725" alt="image" src="https://github.com/user-attachments/assets/fa6b0c0e-6a08-4431-b7b4-8e05f21adcd5" />

## Summary by Sourcery

Enforce valid non-negative cancellation fee ranges across order and event cancellation workflows.

Bug Fixes:
- Prevent negative cancellation fees from being accepted in order and event cancellation forms.

Enhancements:
- Constrain percentage-based cancellation fees to the valid 0–100 range.

Tests:
- Add coverage for rejecting negative order and event cancellation fee values.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
  - Prevented negative cancellation fees from being accepted.
  - Ensured fixed and per-ticket cancellation fees cannot be negative.
  - Restricted percentage-based cancellation fees to values between 0% and 100%.
  - Preserved order totals when invalid cancellation fees are submitted.
  - Improved validation messages for invalid cancellation fee values.

- **Tests**
  - Added coverage confirming invalid cancellation fees are rejected and valid fee values are accepted.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: app/tests/tickets/control/test_orders.py [hunk 1]
```diff
@@ -931,6 +931,43 @@ def test_order_cancel_unpaid_no_fees_allowed(client, env):
...
+        o.save()
+    client.login(email='dummy@dummy.dummy', password='dummy')
+    client.get('/control/event/dummy/dummy/orders/FOO/transition?status=c')
+    client.post(
+        '/control/event/dummy/dummy/orders/FOO/transition',
+        {'status': 'c', 'cancellation_fee': '-5.00'},
+    )
+    with scopes_disabled():
+        o = Order.objects.get(id=env[2].id)
+        assert o.positions.exists()
```
[the remark this answers is not in the document]
Comment (ArnavBallinCode) [reply app/tests/tickets/control/test_orders.py:951]: address comment @anuragverma025

**Review by ArnavBallinCode — CHANGES_REQUESTED** [review]
Please address remaining ai comments

**Review by Rachit7168 — APPROVED** [review]
LGTM , Tested

**Review by ArnavBallinCode — APPROVED** [review]
LGTM
